templates/Carbon/etc/fail2ban/jail.conf.erb

```diff
@@ -187,7 +187,7 @@ action = %(<%= scope['::fail2ban::action'] %>)s
 
 [sshd]
 enabled = <%= scope['::fail2ban::jails'].include? "sshd" %>
-port    = ssh
+port    = <%= scope['::fail2ban::jail_ports'].key?('sshd') ? scope['::fail2ban::jail_ports']['sshd'] : 'ssh' %>
```

Comment:
The scope method is deprecated, can you instead use [this style](https://github.com/voxpupuli/puppet-fail2ban/pull/60/files#diff-5cca39dfcde3749235d7e9f95ff097cdR25) or convert the template to epp?

Instead of using so many ternary statements we should use a defaults hash, merge that once with `$jail_ports` and afterwards directly access the hash in the templates